File: backend-java/src/main/java/com/skewwhiffy/auraltester/dao/QuestionDao.java
package com.skewwhiffy.auraltester.dao;

import com.skewwhiffy.auraltester.dto.question.QuestionType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;

import java.util.UUID;

@Entity(name = "question")
public class QuestionDao {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Getter
    private UUID id;
    @Getter
    private QuestionType questionType;
    @Getter
    private String question;

    QuestionDao() {
    }

    public QuestionDao(QuestionType questionType, String question) {
        this.questionType = questionType;
        this.question = question;
    }
}

File: backend/src/test/kotlin/com/skewwhiffy/auraltester/dao/QuestionDao.kt
package com.skewwhiffy.auraltester.dao

import com.skewwhiffy.auraltester.dto.question.QuestionType
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import java.util.UUID

@Entity(name = "question")
data class QuestionDao(
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: UUID? = null,
    var questionType: QuestionType,
    var question: String
)
